arch/arm/src/gd32f4/gd32f4xx_i2c.c

```diff
@@ -0,0 +1,2731 @@
+   * trace table accordingly.
+   */
+
+  gd32_i2c_tracenew(priv, status);
+  gd32_i2c_traceevent(priv, I2CEVENT_ISR_CALL, 0);
+
+  /* Messages handling (1/2)
+   *
+   * Message handling should only operate when a message has been completely
+   * sent and after the ISR had the chance to run to set bits after the last
+   * written/read byte, i.e. priv->dcnt == -1. This is also the case in when
+   * the ISR is called for the first time. This can seen in
+   * gd32_i2c_process() before entering the gd32_i2c_sem_waitdone() waiting
+   * process.
+   *
+   * Message handling should only operate when:
+   *     - A message has been completely sent and there are still messages
+   *       to send(i.e. msgc > 0).
+   *     - After the ISR had the chance to run to set start bit or
+   *       termination flags after the last written/read byte(after last byte
+   *       dcnt=0, msg handling dcnt = -1).
+   *
+   * When the ISR is called for the first time the same conditions hold.
+   * This can seen in gd32_i2c_process() before entering the
+   * gd32_i2c_sem_waitdone() waiting process.
+   */
+
+#ifdef CONFIG_GD32F4_I2C_DMA
+  /* If ISR gets called (ex. polling mode) while DMA is still in
+   * progress, we should just return and let the DMA finish.
+   */
+
+  ctl1 = gd32_i2c_getreg(priv, GD32_I2C_CTL1_OFFSET);
+  if ((ctl1 & I2C_CTL1_DMAON) != 0)
+    {
+#ifdef CONFIG_DEBUG_I2C_INFO
+      size_t left = gd32_dma_tansnum_get(priv->rxdma);
+
+      i2cinfo("DMA in progress: %ld [bytes] remainining. Returning.\n",
+              left);
+#endif
+      return OK;
+    }
+#endif
+
+  if (priv->dcnt == -1 && priv->msgc > 0)
+    {
+      /* Any new message should begin with "Start" condition
+       * However there were 2 situations where that was not true
+       * Situation 1:
+       *    Next message continue transmission sequence of previous message
+       *
+       * Situation 2: If an error is injected that looks like a STOP the
+       * interrupt will be reentered with some status that will be incorrect.
+       * This will ensure that the error handler will clear the interrupt
+       * enables and return the error to the waiting task.
+       */
+
+      if (((priv->msgv[0].flags & I2C_M_NOSTART) != 0 &&
+           (status & I2C_STAT0_TBE) == 0) ||
+          ((priv->msgv[0].flags & I2C_M_NOSTART) == 0 &&
+           (status & I2C_STAT0_SBSEND) == 0))
+        {
+#if defined(CONFIG_GD32F4_I2C_DMA) || defined(CONFIG_I2C_POLLED)
+          return OK;
+#else
+          priv->status |= I2C_STAT0_SMBTO;
+          goto state_error;
+#endif
+        }
+
+      i2cinfo("Switch to new message\n");
+
+      /* Get current message to process data and copy to private structure */
+
+      priv->ptr           = priv->msgv->buffer;   /* Copy buffer to private struct */
+      priv->dcnt          = priv->msgv->length;   /* Set counter of current msg length */
+      priv->flags         = priv->msgv->flags;    /* Copy flags to private struct */
```

Comment:
maybe some extra spaces before `=` can be removed here?

Reply:
Thanks，I will modify according to comments.